Client/Client.cpp

```diff
@@ -1,44 +1,109 @@
 #include <iostream>
 #include "Client.hpp"
 
-Client::Client() {}
-
-Client::Client(int port, std::string &name, std::string &nick,
-        std::string &pass) : m_port(port), m_name(name), m_nick(nick), m_pass(pass) {
-
-    m_msg = "";
+Client::Client(const char *port, const char *servaddr)
+    : m_port(port), m_servaddr(servaddr), m_clientsock(-1), m_login(), m_pass() {}
+
+bool Client::Start() {
+    struct addrinfo clientaddr, *result;
+
+    memset(&clientaddr, 0, sizeof(clientaddr));
+    clientaddr.ai_family = AF_UNSPEC;
+    clientaddr.ai_socktype = SOCK_STREAM;
+
+    if (getaddrinfo(m_servaddr, m_port, &clientaddr, &result) < 0) {
+        std::cerr << "getaddrinfo error" << std::endl;
+        return false;
+    }
+
+    m_clientsock = socket(result->ai_family, result->ai_socktype, result->ai_protocol);
+    if (m_clientsock < 0) {
+        std::cerr << "Failed to create the socket" << std::endl;
+        freeaddrinfo(result);
+        return false;
+    }
+
+    if (connect(m_clientsock, result->ai_addr, result->ai_addrlen) < 0) {
+        std::cerr << "Failed to connect to the server" << std::endl;
+        close(m_clientsock);
+        freeaddrinfo(result);
+        return false;
+    }
+
+    freeaddrinfo(result);
+    return true;
 }
 
-void Client::Run() {}
-
-void Client::SignUp() {}
-
-void Client::SignIn() {}
-
-int Client::getPort() const { return m_port; }
-
-std::string Client::getName() const { return m_name; }
-
-std::string Client::getNick() const { return m_nick; }
-
-std::string Client::getPass() const { return m_pass; }
-
-std::string Client::getMsg() const { return m_msg; }
-
-void Client::setName(std::string &name) { m_name = name; }
-
-void Client::setNick(std::string &nick) { m_nick = nick; }
-
-void Client::setPass(std::string &pass) { m_pass = pass; }
+bool Client::SendMsgToServer(const std::string &msg) {
```

Comment:
you can use template, for example:
```
template <typename T>
bool send_data(T data, size_t size)
{
    if(send(device_socket,  data, size, 0) < 0)
    {
        ERROR("send error: %d\n", errno);
        return false;
    }

    return true;
}
```
in the future you will use other buffers to send